### ISSUE TITLE
All content results scope

### ISSUE THREAD

**Issue by panchovm** [WordPress/openverse-frontend#1541]
## Assigned reviewers
- [x] @zackkrida 
- [x] @krysal   

## Description

The existing "All content" results page on mobile breakpoints, `sm` and smaller, have limited space for displaying information, and therefore, interacting with each content type item adds UX complexity and risks overloading the page with information and interfaces. On the other hand, we do not know exactly how much value this layout adds to the searching experience as we do not collect user data and have not called for feedback.

Given the above, the page was designed to be an in-between space for getting a glance at the variety of openverse results, where each item is displayed in its basic form, and to bring brand elements, more colors presence and the logo's repetition effect in the box grid. It is not like the content-type result view where visitors can interact with an element through suitable features. This layout reuses the box grid from its desktop version, and from there, visitors jump to another content type through the switcher or see the item details on the single result view.

Nonetheless, we have discussed allowing visitors to interact with each item as they do on bigger breakpoints. In short, expanding the page scope. This issue is to discuss and review the page goal, see the edge cases, possible concerns, and envision new content type integrations (i.e., video). Your visions are crucial for this purpose.

Here is a quick prototype I made to test the existing content types interactions and thinking of twearks for consistency. I could not reproduced some behaviors but you have an idea.

https://user-images.githubusercontent.com/895819/174649076-5185c487-3d4a-47bc-b39a-a27822126dbd.mp4

A few interaction questions have come to my mind while assembling the prototype:
1. If you play an audio track and then tap the rotation button of a 3D model item, does the audio stop and the global player close?
2. If I am playing an audio and then open an image to see its single result view, does the player stop?
3. How do we envision the video interaction in this view without leaving the page?

I personally like the idea of interacting with content on the All content results page, but my main concern is overloading the page with elements and building a non-sustainble view. I still agree with highlighting brand elements and treating this view as a gallery/mosaic where we can introduce openverse; even embedding it in other places. But the brand and content interaction balance is tricky as we always lean more over one side.

We do not need to decide now and we can ended up keeping a simple version, just allowing going to the single result view. Eventually we can make a public call for feedback to discover what visitors value most from this view and complement this discussion with those inputs.

**Comment by obulat** [comment]
I think I like the idea of showing off all that Openverse has to offer _especially_ on mobile screens because so many people only have access to the internet on mobiles. Although it does open the issue of performance, and rendering in various apps' internal browsers (I'm mainly thinking about opening a link to Openverse from inside the social media apps). But that is of course out of scope for this issue.

As for your questions, @panchovm :
> If you play an audio track and then tap the rotation button of a 3D model item, does the audio stop and the global player close?
I don't think that 3D model rotation should interfere with the audio sound. Unless its a landscape mode on the mobile, for which I don't even know how the global audio player looks, to be honest.

>  If I am playing an audio and then open an image to see its single result view, does the player stop?
I think the global audio player should be displayed on the 'All Content' and 'Single Audio result' pages. On other pages it should close. On the other hand, if someone is using the player to listen to music while searching for other media types, that would be a nuisance... I don't know, do you think this scenario is a common one?

> How do we envision the video interaction in this view without leaving the page?
This is a really tough one. Especially considering that we have square items in the 'All content' view, and the videos are usually rectangular. I have no idea

**Comment by panchovm** [comment]
Thanks for replying.

> I don't think that 3D model rotation should interfere with the audio sound

My only concern with this interaction is overwhelming the page with states and things moving simultaneously, like "a busy page."

> if someone is using the player to listen to music while searching for other media types, that would be a nuisance... I don't know, do you think this scenario is a common one?

I have neither clues nor data to select an option, but I am not dismissing it either. If we decide to add a music listening experience in the future, we might need to open the discussion to include features like a queue list, a full-screen player view, next and previous track features, and so on. I like the idea, but for now, I would not spicy the scenario with more complexity.

For previwing videos on the all content page, now I am considering displaying frames from the video in a similar way of previewing 3D model content. Youtube used to do that but I now see a mini-player playing the video on `hover`.

**Comment by zackkrida** [comment]
Thank you for writing this! The main thing I want to say is that I am happy to continue supporting the 'all content' view and providing interactive previews for each content type. The page does provide a very good visual representation of everything Openverse has to offer. Because the page is 'static' by default, meaning that things only move around or media only plays based on user interaction, I do not worry about the page being too busy. 

As for some of the specific questions:

A few interaction questions have come to my mind while assembling the prototype:

> If you play an audio track and then tap the rotation button of a 3D model item, does the audio stop and the global player close?

I think the audio can continue to play, since the 3d models will not have sound of their own. 


> If I am playing an audio and then open an image to see its single result view, does the player stop?

The question here is essentially whether or not we make the global audio player truly global or not. I think we need a standalone conversation about that, could you make a separate issue? For example, we need to finalize if and when the global player displays on desktop and exactly on which pages, as you've mentioned here.

> How do we envision the video interaction in this view without leaving the page?

I think we can worry about that when we get to video, but we could do something similar to 3D models but with a play button, where a quick low-resolution snippet of the video plays inline. You can see something similar on other stock video sites:

https://www.pexels.com/videos/

> Eventually we can make a public call for feedback to discover what visitors value most from this view and complement this discussion with those inputs.

💯

**Comment by panchovm** [comment]
Coming back here to push a conclusion to this discussion.

Sharing my thoughts on one point shared by @zackkrida 

> The question here is essentially whether or not we make the global audio player truly global or not. I think we need a standalone conversation about that

I would like to talk about this, especially the use case on desktop and how the global player would work. Nonetheless, we are addressing other core components in the forthcoming releases. With that in mind, this topic can wait until we move out of the beta status and discuss ways of content interaction.

I can not recall whether I assigned reviewers or not, so pinging @krysal to follow the template procedure and wrap up this issue with a summary comment.

**Comment by krysal** [comment]
I like the idea of maintaining our "All content results" view! As long as it remains feasible, given the number of different media types we plan to add.

> 1. If you play an audio track and then tap the rotation button of a 3D model item, does the audio stop and the global player close?

I don't think it would interfere either, but I suppose that if we click to rotate a second 3D model, then the first one will stop? And will we allow it to rotate arbitrarily with the user mouse + click? It's the most common interaction, but in such reduced space may be tricky to achieve it and probably not worth the effort in making that feature.

> 2. If I am playing an audio track and then open an image to see its single result view, does the player stop?

I agree here with @obulat on keeping it limited to the "All Content" and 'Single Audio result' pages, at least for the full launch of Audio. We can consider expanding the global player for future iterations but generally, I like to _keep it simple_.

--- 

TLDR; Scope approved ✅

**Comment by panchovm** [comment]
Closing this issue ✨
